### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by ericpre**
### Progress of the PR
- [x] Fix scipy deprecation warnings,
- [x] Fix numpy deprecation warning,
- [x] Use new and recommended numpy polynomial API,
- [x] Use hyperspy public API in favour of private API, 
- [n/a] docstring updated (if appropriate),
- [n/a] update user guide (if appropriate),
- [n/a] added tests,
- [x] add a changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/exspy/blob/main/upcoming_changes/README.rst)),
- [x] Check formatting of the changelog entry (and eventual user guide changes) in the `docs/readthedocs.org:exspy` build of this PR (link in github checks)
- [x] ready for review.

### CODE REVIEW

**Comment by ericpre** [comment]
This should fix the failures in https://github.com/hyperspy/hyperspy-extensions-list/actions/runs/8494826786/job/23270443981.

File: exspy/signals/eels.py [hunk 1]
```diff
@@ -622,10 +620,10 @@ def estimating_function(data, threshold=None):
                     if binned:
                         return data.sum()
                     else:
-                        from scipy.integrate import simps
+                        from scipy.integrate import simpsom
```
Comment (jlaehne) [inline exspy/signals/eels.py:623]: The integration rule should be `simpson` and not `simpsom`!
Obviously, the line is not included in tests if this did not yield a failure.

File: exspy/signals/eels.py [hunk 2]
```diff
@@ -622,10 +620,10 @@ def estimating_function(data, threshold=None):
...
 
                         axis = ax.axis[:ind]
-                        return simps(y=data, x=axis)
+                        return simpsom(y=data, x=axis)
```
Comment (jlaehne) [inline exspy/signals/eels.py:626]: as above

File: exspy/signals/eels.py [hunk 3]
```diff
@@ -25,16 +25,14 @@
 from scipy import constants
 from prettytable import PrettyTable
 
+import hyperspy
```
Comment (jlaehne) [inline exspy/signals/eels.py:28]: Why import the whole `hyperspy` - I only see lines calling the api imported in the next line

File: exspy/misc/eels/tools.py [hunk 4]
```diff
@@ -109,29 +107,30 @@ def estimate_variance_parameters(
     """Find the scale and offset of the Poissonian noise
 
     By comparing an SI with its denoised version (i.e. by PCA),
-    this plots an
-    estimation of the variance as a function of the number of counts
-    and fits a
-    polynomy to the result.
+    this plots an estimation of the variance as a function of the number of counts
+    and fits a polynomy to the result.
```
Comment (jlaehne) [inline exspy/misc/eels/tools.py:111]:
```suggestion
    and fits a polynomial to the result.
```

File: exspy/misc/eels/tools.py [hunk 5]
```diff
@@ -60,16 +60,14 @@ def _estimate_gain(
         variance2fit = variance
         average2fit = average
 
-    fit = np.polyfit(average2fit, variance2fit, pol_order)
+    fit = np.polynomial.Polynomial.fit(average2fit, variance2fit, pol_order)
```
Comment (jlaehne) [inline exspy/misc/eels/tools.py:63]: The whole function seems to be missing a test as all touched lines raise a warning.

File: exspy/components/eels_vignetting.py [hunk 6]
```diff
@@ -46,7 +46,7 @@ def __init__(self):
         self.right.value = np.nan
         self.side_vignetting = False
         self.fix_side_vignetting()
-        self.gaussian = Gaussian()
+        self.gaussian = hs.model.components1D.Gaussian()
```
Comment (jlaehne) [inline exspy/components/eels_vignetting.py:49]: no test for this component?

**Comment by ericpre** [comment]
For the missing tests, I had a look and I have no idea, what these functions are supposed to do! I would suggest to leave it for another PR which will need help from @francisco-dlp.

**Comment by ericpre** [comment]
@jlaehne, do you want to have another look at this PR?
I would be quite keen to finish the scipy/numpy deprecation and get them out of the way! :)

**Review by jlaehne — APPROVED** [review]

**Pull request merged**